optimism/src/lookups.rs

```diff
@@ -7,34 +7,34 @@ use kimchi::{
         constants::{RATE_IN_BYTES, ROUNDS},
         Keccak, RC,
     },
-    o1_utils::Two,
+    o1_utils::{FieldHelpers, Two},
 };
 use kimchi_msm::{LookupTableID, MVLookupTable};
 
 /// All of the possible lookup table IDs used in the zkVM
-#[derive(Copy, Clone, Debug)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
 pub enum LookupTableIDs {
-    // RAM Tables
-    MemoryLookup = 0,
-    RegisterLookup = 1,
-    /// Syscalls communication channel
-    SyscallLookup = 2,
-    /// Input/Output of Keccak steps
-    KeccakStepLookup = 3,
-
-    // Read Tables
+    // Read tables come first to allow indexing with the table ID for the multiplicities
     /// Single-column table of all values in the range [0, 2^16)
-    RangeCheck16Lookup = 4,
+    RangeCheck16Lookup = 0,
```

Comment:
Is it dangerous to have a table with ID 0 which contains the value 0?

Reply:
You mean because no matter what the joint combiner could be, this entry will always be zero? That's an interesting point! I guess a simple fix for this is just reorder the ids so that the first entry of the 0 table contains a nonzero value 👍🏻